api/src/main/java/com/okta/sdk/resource/identity/provider/MicrosoftIdentityProviderBuilder.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.resource.identity.provider;
+
+import com.okta.commons.lang.Classes;
+
+public interface MicrosoftIdentityProviderBuilder extends IdentityProviderBuilder<MicrosoftIdentityProviderBuilder> {
```

Comment:
Are these interfaces needed?
It doesn't look like there are anything specific here? Maybe something like:

```java
public class IdentityProviderBuilders {


    public static OIDCIdentityProviderBuilderoidc() { ... }

    public static IdentityProviderBuilder facebook() { ... }
  
    public static IdentityProviderBuilder google() { ... }
...

}
```

Reply:
Updated.